### PR TITLE
ADEN-4683 | add missing unit tests for resolved state resolution

### PR DESCRIPTION
**Pull request by mleczey**
## Links:
- https://wikia-inc.atlassian.net/browse/ADEN-4683

### CODE REVIEW

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 1]
```diff
@@ -2,128 +2,140 @@
...
+					src: DEFAULT_IMAGE
+				},
+				defaultStateSrc: BIG_IMAGE,
+				resolvedStateSrc: RESOLVED_IMAGE
+			},
+			image2: {
+				element: {
+					src: DEFAULT_IMAGE
+				},
+				defaultStateSrc: BIG_IMAGE_2,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "BIG_IMAGE_2" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 2]
```diff
@@ -2,128 +2,140 @@
...
+				resolvedStateSrc: RESOLVED_IMAGE_2
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "RESOLVED_IMAGE_2" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 3]
```diff
@@ -2,128 +2,140 @@
 describe('ext.wikia.adEngine.slot.resolvedState', function () {
 	'use strict';
 
-	var BIG_IMAGE = 'bigImage.png',
+	function createCorrectParams() {
+		return {
+			imageSrc: BIG_IMAGE,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "BIG_IMAGE" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 4]
```diff
@@ -2,128 +2,140 @@
...
+			aspectRatio: ASPECT_RATIO,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "ASPECT_RATIO" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 5]
```diff
@@ -2,128 +2,140 @@
...
+			resolvedState: {
+				aspectRatio: RESOLVED_STATE_ASPECT_RATIO,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "RESOLVED_STATE_ASPECT_RATIO" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 6]
```diff
@@ -2,128 +2,140 @@
...
+				imageSrc: RESOLVED_IMAGE
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "RESOLVED_IMAGE" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 7]
```diff
@@ -2,128 +2,140 @@
...
+			},
+			backgroundImage: {
+				src: DEFAULT_IMAGE
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Move the declaration of "DEFAULT_IMAGE" before this usage. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MAJOR 7 major

Watch the comments in this conversation to review them.

**Review by v3rth — APPROVED** [review]
LGTM

**Pull request merged**